packages/compass-aggregations/src/components/aggregation-side-panel/stage-wizard-use-cases/match/group.tsx

```diff
@@ -0,0 +1,395 @@
+    if (!groups[groupIdx]) {
+      return;
+    }
+
+    const remainingGroups = [...groups];
+    remainingGroups.splice(groupIdx, 1);
+    onGroupsChange(remainingGroups);
+  },
+
+  handleAddGroupClick() {
+    if (groups.length > 1) {
+      // There can only be two groups ($and and $or)
+      // at any nesting level.
+      return;
+    }
+
+    const existingGroup = groups[0];
+    const alreadySelectedOperator = existingGroup?.logicalOperator;
+    const newGroupOperator =
+      alreadySelectedOperator === '$and' ? '$or' : '$and';
+    onGroupsChange([
+      ...groups,
+      createGroup({ logicalOperator: newGroupOperator }),
+    ]);
+  },
+});
+
+// Components - GroupHeader
+const groupHeaderStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-start',
+  gap: spacing[3],
+  width: `calc(100% - ${CONDITION_CONTROLS_WIDTH})`,
+});
+
+export const GroupHeader = ({
+  dataTestId,
+  disableAddGroup,
+  disableRemoveGroup,
+  operator,
+  onOperatorChange,
+  onAddGroupClick,
+  onRemoveGroupClick,
+}: GroupHeaderProps) => {
+  return (
+    <div data-testid={dataTestId} className={groupHeaderStyles}>
+      <SegmentedControl
+        size="small"
+        value={operator}
+        onChange={(operator) => {
+          onOperatorChange(operator as LogicalOperator);
+        }}
+      >
+        <SegmentedControlOption value="$and">AND</SegmentedControlOption>
+        <SegmentedControlOption value="$or">OR</SegmentedControlOption>
+      </SegmentedControl>
+      <div>
+        <IconButton
+          disabled={disableAddGroup}
+          aria-label="Add Group"
+          onClick={onAddGroupClick}
+        >
+          <Icon glyph="Plus" />
+        </IconButton>
+        <IconButton
+          disabled={disableRemoveGroup}
+          aria-label="Remove Group"
+          onClick={onRemoveGroupClick}
+        >
+          <Icon glyph="Trash" />
+        </IconButton>
+      </div>
+    </div>
+  );
+};
+
+// Components - GroupFooter
+const groupFooterStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-end',
+  // This is done to align the footer's end with the end of last field in the
+  // condition right above it spacing[2] is the padding provided to conditions
+  // container
+  width: `calc(100% - ${CONDITION_CONTROLS_WIDTH + spacing[2]}px)`,
+});
+
+const GroupFooter = ({
+  dataTestId,
+  onAddNestedGroupClick,
+}: GroupFooterProps) => {
+  return (
+    <div data-testid={dataTestId} className={groupFooterStyles}>
+      <Button
+        variant="default"
+        size="xsmall"
+        type="button"
+        onClick={onAddNestedGroupClick}
+      >
+        ADD GROUP
+      </Button>
+    </div>
+  );
+};
+
+// Components - Group
+const groupStyles = css({
+  display: 'flex',
+  gap: spacing[3],
+  flexDirection: 'column',
+});
+
+const conditionContainerStyles = cx(
+  groupStyles,
+  css({
+    paddingRight: spacing[2],
+  })
+);
+
+const nestedGroupsContainerStyles = (groupsLength: number) =>
```

Comment:
Calling css inside render is an anti-pattern that can potentially cause performance issues, [we spent some time cleaning it up recently](https://github.com/mongodb-js/compass/pull/4156), I'd suggest you refactor this too. If styles are mostly static, you can define them outside a render and switch on variable, if they are fully dynamic, it is better to use `style` prop

Reply:
Thank you @gribnoysup for bringing my attention to this. I have addressed this in my recent commits.